test/graphkb.test.js

```diff
@@ -0,0 +1,183 @@
+const { simplifyRecordsLinks, shouldUpdate } = require('../src/graphkb');
+
+describe('shouldUpdate', () => {
+    test('test for disease object', () => {
+        const model = 'disease';
+        const originalContent = {
+            '@class': 'Disease',
+            '@rid': '#133:8',
+            alias: true,
+            createdAt: 1565314461881,
+            createdBy: '#29:0',
+            deprecated: false,
+            description: 'congenital abnormality characterized by the presence of only one kidney.',
+            displayName: 'congenital single kidney [c101220]',
+            history: '#135:28899',
+            in_AliasOf: [
+                '#67:21022',
+                '#66:23',
+            ],
+            name: 'congenital single kidney',
+            out_AliasOf: [
+                '#66:31991',
+            ],
+            source: '#40:3',
+            sourceId: 'c101220',
+            updatedAt: 1594438640025,
+            updatedBy: '#29:0',
+            url: 'http://ncicb.nci.nih.gov/xml/owl/evs/thesaurus.owl#c101220',
+            uuid: '709eb34b-27ff-42f5-be0c-9051c639deb0',
+        };
+        originalContent.source = {
+            '@class': 'Source',
+            '@rid': '#40:3',
+            createdAt: 1565314457745,
+            createdBy: '#29:0',
+            description: 'nci thesaurus (ncit) provides reference terminology for many nci and other systems. it covers vocabulary for clinical care, translational and basic research, and public information and administrative activities.',
+            displayName: 'NCIt',
+            longName: 'nci thesaurus',
+            name: 'ncit',
+            sort: 2,
+            updatedAt: 1565314457745,
+            updatedBy: '#29:0',
+            url: 'https://ncit.nci.nih.gov/ncitbrowser',
+            usage: 'https://creativecommons.org/licenses/by/4.0',
+            uuid: 'dad84739-b1e3-4686-b055-6bc3c3de9bc3',
+        };
+        const newContent = { ...originalContent };
+        newContent.name = 'a new name';
+        newContent.displayName = 'a new display name';
+        const excludedFieldsExhaustive = ['name', 'displayName'];
+        const excludedFieldsNonExhaustive = ['name'];
+
+        expect(shouldUpdate(
+            model,
+            originalContent,
+            newContent,
+            excludedFieldsNonExhaustive,
+        )).toBe(true);
+
+        expect(shouldUpdate(
+            model,
+            originalContent,
+            newContent,
+            excludedFieldsExhaustive,
+        )).toBe(false);
+
+        newContent.source.name = 'a new source name';
+        expect(shouldUpdate(
+            model,
+            originalContent,
+            newContent,
+            excludedFieldsExhaustive,
+        )).toBe(false);
+    });
+
+    test('test for statement object', () => {
```

Comment:
It would be good to make this name make it more obvious what you are testing, I might even nest this in a second describe block. ex.

```
describe('shouldUpdate', () => {
  describe('statement', () => {
    test('true when non-excluded field changes');
    test('false when changed field is excluded');
    test('false when same object passed and no fields excluded');
 })
})
```

I would also make sure you have a test for when a linked record changes (should never cause update to the current record)

Reply:
Ok, didn't know I could nest describe function, will do the changes. I did made a test for linked record changes though.